pkg/expr/func_num.go

```diff
@@ -112,3 +115,24 @@ func average(v ...interface{}) float64 {
 
 	return sum / j
 }
+
+func random(v ...float64) (out float64, err error) {
+	var from, to float64
+
+	switch len(v) {
+	case 1:
+		if to = v[0]; to < 0 {
+			return 0, errors.New("unexpected input type")
+		}
+	case 2:
+		if from, to = v[0], v[1]; from < 0 || to < 0 {
+			return 0, errors.New("unexpected input type")
+		}
```

Comment:
I would do this with multiple ifs to avoid repeating logic.
Something in the lines of:

```
if len(v) > 0 {
  if v[0] < 0 { error... }
}
if len(v) > 1 {
  if v[1] < 0 {error ... }
}
...
```

Personal preference; feel free to ignore.